### PR TITLE
Finalize Product & Order API as well as post purchase to Squizz

### PR DESCRIPTION
**Pull request by lemon037**

### CODE REVIEW

**Comment by lemon037** [comment]
The old create(submit) order method, does not store the real order information, instead, it stores hardcoded data. Besides, the old submit order accepts all the information from the frontend without any checking, which should not be allowed.

**Comment by hpy9707** [comment]
As for me it is a little bit weird that check the validation in the backend. it should be checked in front end. But it is not a big problem. I only change one thing: in the util folder I change from "validation"into "Validation"

**Comment by lemon037** [comment]
> As for me it is a little bit weird that check the validation in the backend. it should be checked in front end. But it is not a big problem. I only change one thing: in the util folder I change from "validation"into "Validation"

I mean the backend cannot store the data without checking their correctness, also the frontend should take charge of the format of the data to be posted.

File: app/Resource/OrderResource.py [hunk 1]
```diff
@@ -224,4 +225,51 @@ def get_order_history(self, session_id):
                 'message': 'Exception occurred when retrieving order history'
             }
 
-        return result
\ No newline at end of file
+        return result
+
+    def create_order(self, org: Organization, customer: Customer, delivery: Address, billing: Address,
```
Comment (ansabkhaliq) [inline app/Resource/OrderResource.py:230]: I think this method does what Store_purchase method was doing. Am I right? If yes, then delete the old method

File: app/Service/OrderService.py [hunk 2]
```diff
@@ -29,3 +38,42 @@ def get_order_history(session_id) -> dict:
         }
 
     return result
+
+
+def save_order(session_key, customer_id, delivery_addr_id, billing_addr_id, lines, instructions=""):
```
Comment (ansabkhaliq) [inline app/Service/OrderService.py:43]: This method seems to be doing what Submit_order was doing.... I would say if that is the case, please get rid of the old method

File: app/Service/SquizzGatewayService.py [hunk 3]
```diff
@@ -154,3 +159,37 @@ def submit_purchase(self, session_id, order_details) -> Tuple[bool, Optional[lis
         data = self.requests.post(purchaseURL, json=result, headers=header).json()
         # return the result code of purchase
         return data["result_code"], Order(parameter)
+
+    def submit_order(self, org: Organization, customer: Customer, order_details_list: [OrderDetail]):
```
Comment (ansabkhaliq) [inline app/Service/SquizzGatewayService.py:163]: I think once again, submit_order is doing what  old submit_purchase was doing.. if that is the case, plz get rid of the older method.

**Review by ansabkhaliq — CHANGES_REQUESTED** [review]

**Review by ansabkhaliq — APPROVED** [review]

**Pull request merged**